ui/lib/core/addon/utils/client-count-utils.js

```diff
@@ -125,89 +125,28 @@ export const sortMonthsByTimestamp = (monthsArray) => {
 
 export const namespaceArrayToObject = (totalClientsByNamespace, newClientsByNamespace, month, timestamp) => {
   if (!totalClientsByNamespace) return {}; // return if no data for that month
-  // all 'new_client' data resides within a separate key of each month (see data structure below)
-  // FIRST: iterate and nest respective 'new_clients' data within each namespace and mount object
-  // note: this is happening within the month object
-  const nestNewClientsWithinNamespace = totalClientsByNamespace?.map((ns) => {
-    const newNamespaceCounts = newClientsByNamespace?.find((n) => n.label === ns.label);
-    if (newNamespaceCounts) {
-      const newClientsByMount = [...newNamespaceCounts.mounts];
-      const nestNewClientsWithinMounts = ns.mounts?.map((mount) => {
-        const new_clients = newClientsByMount?.find((m) => m.label === mount.label) || {};
-        return {
-          ...mount,
-          new_clients,
-        };
-      });
-      return {
-        ...ns,
-        new_clients: {
-          label: ns.label,
-          ...destructureClientCounts(newNamespaceCounts),
-          mounts: newClientsByMount,
-        },
-        mounts: [...nestNewClientsWithinMounts],
-      };
-    }
-    return {
-      ...ns,
-      new_clients: {},
-    };
-  });
-  // SECOND: create a new object (namespace_by_key) in which each namespace label is a key
-  const namespaces_by_key = {};
-  nestNewClientsWithinNamespace?.forEach((namespaceObject) => {
-    // THIRD: make another object within the namespace where each mount label is a key
-    const mounts_by_key = {};
-    namespaceObject.mounts.forEach((mountObject) => {
-      mounts_by_key[mountObject.label] = {
-        month,
-        timestamp,
-        ...mountObject,
-        new_clients: { month, ...mountObject.new_clients },
-      };
-    });
-
-    const { label, new_clients } = namespaceObject;
-    namespaces_by_key[label] = {
-      month,
-      timestamp,
-      ...destructureClientCounts(namespaceObject),
-      new_clients: { month, ...new_clients },
-      mounts_by_key,
-    };
-  });
+
+  // namespaces_by_key is used to filter monthly activity data by namespace
+  // it's an object in each month data block where the keys are namespace paths
+  // and values include new and total client counts for that namespace in that month
+  const namespaces_by_key = totalClientsByNamespace.reduce((nsObject, ns) => {
```

Comment:
I purposefully assigned this to a const here that matches the key on the response this gets added to so it's easier to track down in the future  (instead of just returning)